### PR TITLE
leo's completed week 3 assessment

### PR DESCRIPTION
**Pull request by Melaza6**
# Submitting Week 3 Assessment

### ✅ Completed Files

Mark the following as confirmation that the files in this repository are complete.

- [  ✅] Coding Challenges
- [  ✅] Interview Questions
- [ ✅ ] React Debugging

### 📝 Feedback

If you would like specific feedback on a particular problem please leave details here:

My Yarn wouldn't upload package.json and had a hard time running the test. 
functions do work correctly.

### CODE REVIEW

File: code-challenges.test.js [hunk 1]
```diff
@@ -20,8 +20,38 @@ const fibLength1 = 6
 const fibLength2 = 10
 // Expected output: [1, 1, 2, 3, 5, 8, 13, 21, 34, 55]
 
+// describe("fiboSeq", () => {
+  // it("takes in a number (greater than 2) and returns an array that length containing the numbers of the Fibonacci sequence.", () => {
+  //   expect(fiboSeq(fibLength1)).toEqual("[1, 1, 2, 3, 5, 8]")
+  //   expect(fiboSeq(fibLength2)).toEqual("[1, 1, 2, 3, 5, 8, 13, 21, 34, 55]")
```
Comment (Genemichael25) [inline code-challenges.test.js:26]: Nice setup for the test! Just keep in mind what we expect it to equal. Do we need the "" around the array?

File: code-challenges.test.js [hunk 2]
```diff
@@ -20,8 +20,38 @@ const fibLength1 = 6
...
+
 // b) Create the function that makes the test pass.
 
+// pseudo code:
+// Input: a number number that indicates de leghth of the Fibonacci sequence
+// Ouput: the sum of the current index plus the last to come up with the next value of the following index
+// create a function "fiboSeq" that take a number as a parameter
+// create an array "fibo" as the beginning of the sequence
+// in for loop have the iteration start with sum of the first two indexes and .push() to incorporate the sum of the indexes as the value for the next index and so on.
+
```
Comment (Genemichael25) [inline code-challenges.test.js:39]: Nicely done!

File: code-challenges.test.js [hunk 3]
```diff
@@ -49,10 +79,29 @@ const studyMinutesWeek2 = {
...
 // b) Create the function that makes the test pass.
 
-// --------------------3) Create a function that takes in an array and returns an array of the accumulating sum. An empty array should return an empty array.
+// pseudo code:
+// input: takes an array
+// output: an array of the sorted values in an object
+// create a function l2g that takes an array as a parameter
+// use the static method Object.values() on the array to print the values in an array then use built-in method .sort() with given parameter to sort from least to greatest "a-b"
+
```
Comment (Genemichael25) [inline code-challenges.test.js:98]: Awesome pseudocode and very clean code! 👏🏽

File: code-challenges.test.js [hunk 4]
```diff
@@ -64,4 +113,31 @@ const accountTransactions2 = [250, -89, 100, -96]
...
 // b) Create the function that makes the test pass.
+
+// pseudo code:
+// input: takes an array
+// output: the array's accumulated sum as it goes into the next index
+// create a function "trans" that takes an array as a function
+// declared where the index iteration will start
+// using .map() on the arry bring the sum of the values iterated by the next index. 
+
```
Comment (Genemichael25) [inline code-challenges.test.js:134]: Do we need to declare where the iteration will start with .map? Or are we using i = 0 as the initial value?

File: interview-questions.md [hunk 5]
```diff
@@ -8,48 +8,48 @@ Then, research the question to expand on your answer. Even if you feel you have
 
 1. What is JSX? What is one syntax difference between HTML and JSX?
 
-Your answer:
+Your answer: JSX is javaScript lenguage concatenated with HTLM by the React app to effectively write code.
 
-Researched answer:
+Researched answer: JSX is a syntax similart to HTML in react to for easier use by developers. HTML tells the browser how to display its elements.
```
Comment (Genemichael25) [inline interview-questions.md:13]: What is one syntax difference?

File: interview-questions.md [hunk 6]
```diff
@@ -8,48 +8,48 @@ Then, research the question to expand on your answer. Even if you feel you have
...
-Researched answer:
+Researched answer: served as JS functions that return HTML, they are independent and reusable. 
 
 4. What is the difference between state values and props in React?
 
-Your answer:
+Your answer: evetho they are both use to carry information between react components state serves as storage for passing data.
 
-Researched answer:
+Researched answer: props are use to pass data and State manage the data given.
```
Comment (Genemichael25) [inline interview-questions.md:31]: Props can also pass functions! I encourage you to use this space in the researched answers to dive a bit deeper into the concepts and learn something new.

File: react-debugging.html [hunk 7]
```diff
@@ -64,23 +64,26 @@
...
-          <div style={handStyles}>
-            {hand.map((card, index) => {
-              return (
-                <div style={cardStyles} key={index}>
-                  {card}
-                </div>
-              )
-            })}
-          </div>
+          return (
```
Comment (Genemichael25) [inline react-debugging.html:67]: Great job finding the bugs! 🐛

**Review by Genemichael25 — APPROVED** [review]
Great job on this assessment Leo! This shows an Advanced Beginner level of understanding. Keep being as descriptive as possible with the pseudocode and the functions look very clean and concise! One of the things that will help you get to that next level is broadening your answers in interview questions portion. This will help prepare you for interviews, as well as further your understanding of the concepts. Left a few comments for you to look at, keep up the good work!

File: code-challenges.test.js (hunk 4 above)
Comment (Melaza6) [reply code-challenges.test.js:134]: we're using i = 0 as the initial value for the index and to start counting from there.

File: interview-questions.md (hunk 5 above)
Comment (Melaza6) [reply interview-questions.md:13]: Will be looking for a much-researched answer and provide a better answer with syntax examples on this comment threat.

File: interview-questions.md (hunk 6 above)
Comment (Melaza6) [reply interview-questions.md:31]: Will do.

**Pull request merged**